modules/platform/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/PictureViewsGenerationWork.java

```diff
@@ -103,47 +100,24 @@ public void work() {
         setStatus("Extracting");
 
         openSystemSession();
-        if (!session.exists(new IdRef(docId))) {
-            setStatus("Nothing to process");
-            return;
-        }
-
-        DocumentModel workingDocument = session.getDocument(new IdRef(docId));
-        Property fileProp = workingDocument.getProperty(xpath);
-        Blob blob = (Blob) fileProp.getValue();
-        if (blob == null) {
-            // do nothing
-            return;
-        }
 
-        String title = workingDocument.getTitle();
-        setStatus("Generating views");
-        try {
-            PictureResourceAdapter picture = workingDocument.getAdapter(PictureResourceAdapter.class);
-            picture.fillPictureViews(blob, blob.getFilename(), title, null);
-        } catch (DocumentNotFoundException e) {
-            // a parent of the document may have been deleted.
-            setStatus("Nothing to process");
-            return;
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+        getPictureViewsHelper().newTransaction();
+        getPictureViewsHelper().computePictureViews(session, docId, xpath, this::setStatus);
```

Comment:
Why not just creating the PictureViewsHelper here instead of having a transient field and a method?